src/server/mime.ts

```diff
@@ -100,27 +125,26 @@ function createBaseMessage(contentType: string, isBinary = false) {
     return message;
 }
 
-function createEnvelopeKeys(command: string, filename: string,
-                            project: string, responseID: string){
+function createEnvelopeKeys(params: RequestParams){
     let keys: string;
-    if (responseID == null){
+    if (isInitialRequest(params)){
         keys = `<osc-data:keys>
-                        <osc-data:key>${command}</osc-data:key>
-                        <osc-data:key>${project}</osc-data:key>
-                        <osc-data:key>00atf/${filename}</osc-data:key>
+                        <osc-data:key>${params.command}</osc-data:key>
+                        <osc-data:key>${params.project}</osc-data:key>
+                        <osc-data:key>00atf/${params.filename}</osc-data:key>
                     </osc-data:keys>`;
```

Comment:
One thing I had planned to do but I think I never did was to use templates for the xml used in the server documentation. I imagine there's something like [jinja2](https://jinja2docs.readthedocs.io/en/stable/) for Typescript.

Reply:
So you could have an XML template file for each time of message, or chunk of message, and then fill it in with the given parameters.

Reply:
Good idea, I'll make another issue for that! Especially because the indentation is confusing here.